lib/models/device.coffee

```diff
@@ -1811,158 +1832,616 @@ getDeviceModel = (deps, opts) ->
-	# @example
-	# resin.models.device.tags.remove('7cf02a6', 'EDITOR');
-	#
-	# @example
-	# resin.models.device.tags.remove('7cf02a6', 'EDITOR', function(error) {
-	# 	if (error) throw error;
-	# });
-	###
-	exports.tags.remove = tagsModel().remove
+		###*
+		# @summary Get the overriden value of a service variable on a device
+		# @name get
+		# @public
+		# @function
+		# @memberof resin.models.device.serviceVar
+		#
+		# @param {String|Number} uuidOrId - device uuid (string) or id (number)
+		# @param {Number} id - service id
+		# @param {String} key - variable name
+		# @fulfil {String|undefined} - the variable value (or undefined)
+		# @returns {Promise}
+		#
+		# @example
+		# resin.models.device.serviceVar.get('7cf02a6', 123, 'VAR').then(function(value) {
+		# 	console.log(value);
+		# });
+		#
+		# @example
+		# resin.models.device.serviceVar.get(999999, 123, 'VAR').then(function(value) {
+		# 	console.log(value);
+		# });
+		#
+		# @example
+		# resin.models.device.serviceVar.get('7cf02a6', 123, 'VAR', function(error, value) {
+		# 	if (error) throw error;
+		# 	console.log(value)
+		# });
+		###
+		get: (uuidOrId, serviceId, key, callback) ->
+			callback = findCallback(arguments)
+
+			exports.get(uuidOrId, $select: 'id').get('id')
+			.then (deviceId) ->
+				pine.get
+					resource: 'device_service_environment_variable'
+					options:
+						$filter:
+							service_install:
+								$any:
+									$alias: 'si',
+									$expr: si:
+										device: deviceId
+										service: serviceId
+							name: key
+			.then (results) ->
+				if (results[0])
+					results[0].value
+			.asCallback(callback)
+
+		###*
+		# @summary Set the overriden value of a service variable on a device
+		# @name set
+		# @public
+		# @function
+		# @memberof resin.models.device.serviceVar
+		#
+		# @param {String|Number} uuidOrId - device uuid (string) or id (number)
+		# @param {Number} id - service id
+		# @param {String} key - variable name
+		# @param {String} value - variable value
+		# @returns {Promise}
+		#
+		# @example
+		# resin.models.device.serviceVar.set('7cf02a6', 123, 'VAR', 'override').then(function() {
+		# 	...
+		# });
+		#
+		# @example
+		# resin.models.device.serviceVar.set(999999, 123, 'VAR', 'override').then(function() {
+		# 	...
+		# });
+		#
+		# @example
+		# resin.models.device.serviceVar.set('7cf02a6', 123, 'VAR', 'override', function(error) {
+		# 	if (error) throw error;
+		# 	...
+		# });
+		###
+		set: (uuidOrId, serviceId, key, value, callback) ->
+			Promise.try ->
+				value = String(value)
+
+				deviceFilter = if isId(uuidOrId)
+					uuidOrId
+				else
+					$any:
+						$alias: 'd'
+						$expr: d:
+							uuid: uuidOrId
+
+				pine.get
+					resource: 'service_install'
+					options:
+						$filter:
+							device: deviceFilter
+							service: serviceId
+				.tap (serviceInstalls) ->
+					if isEmpty(serviceInstalls)
+						throw new errors.ResinServiceNotFound(serviceId)
+					else if serviceInstalls.length > 1
```

Comment:
NIT: How about dropping the `else` to make it slightly more readable (since this is after a throw)?

Reply:
Sure, done